src/sqlfluff/core/rules/base.py

```diff
@@ -416,6 +435,59 @@ def raw_segments(self):
         )
 
 
+class CrawlBehavior:
+    """Implements how a lint rule traverses the parse tree."""
+
+    def __init__(
+        self, works_on_unparsable: bool, recurse_into: bool, needs_raw_stack: bool
+    ):
+        self.works_on_unparsable = works_on_unparsable
+        self.recurse_into = recurse_into
+        self.raw_stack: Optional[List[RawSegment]]
+        if needs_raw_stack:
+            self.raw_stack = []
+        else:
+            self.raw_stack = None  # pragma: no cover
```

Comment:
"no cover" for now. Will be used in a subsequent PR, when we update rules to take advantage of this capability.